apps/taxes/src/modules/taxjar/taxjar-webhook.service.ts

```diff
@@ -2,44 +2,65 @@ import { AuthData } from "@saleor/app-sdk/APL";
 import {
   OrderCancelledEventSubscriptionFragment,
   OrderConfirmedSubscriptionFragment,
-  TaxBaseFragment,
 } from "../../../generated/graphql";
 import { Logger, createLogger } from "../../lib/logger";
+import { CalculateTaxesPayload } from "../../pages/api/webhooks/checkout-calculate-taxes";
 import { ProviderWebhookService } from "../taxes/tax-provider-webhook";
 import { TaxJarCalculateTaxesAdapter } from "./calculate-taxes/taxjar-calculate-taxes-adapter";
+import { TaxJarClientLogger, createTaxJarClientLogger } from "./logs/taxjar-client-logger";
 import { TaxJarOrderConfirmedAdapter } from "./order-confirmed/taxjar-order-confirmed-adapter";
 import { TaxJarClient } from "./taxjar-client";
 import { TaxJarConfig } from "./taxjar-connection-schema";
-import { CalculateTaxesPayload } from "../../pages/api/webhooks/checkout-calculate-taxes";
 
 export class TaxJarWebhookService implements ProviderWebhookService {
   client: TaxJarClient;
   private logger: Logger;
   private config: TaxJarConfig;
+  private clientLogger: TaxJarClientLogger;
+  private authData: AuthData;
 
-  constructor(
-    config: TaxJarConfig,
-    private authData: AuthData,
-  ) {
+  constructor({
+    configurationId,
+    config,
+    authData,
+  }: {
+    configurationId: string;
+    config: TaxJarConfig;
+    authData: AuthData;
+  }) {
     const taxJarClient = new TaxJarClient(config);
 
     this.client = taxJarClient;
     this.config = config;
+    this.authData = authData;
+
+    this.clientLogger = createTaxJarClientLogger({
+      authData,
+      configurationId,
+    });
```

Comment:
IMO you should inject this logger via constructor (classic dependency injection). 

This is a code smell - you require class to accept configuration id just to allow it to create another class.

Instead, just create logger outside and pass instance here

its also easier to test